vgen/src/main/java/io/gapi/vgen/csharp/CSharpContextCommon.java

```diff
@@ -92,18 +100,21 @@ public String underscoresToCamelCase(
   // This function is necessary to provide a static entry point for the same-named
   // member function.
   public static String s_underscoresToCamelCase(
-      String input, boolean capNextLetter, boolean preservePeriod) {
+      String input, boolean capNextLetter, boolean preservePeriod, boolean forceLower) {
```

Comment:
What does forceLower mean in this case? Basically *only* use _ to create upper-case characters? Any reason the caller can't just lower-case the input first?

Is this method still a port of csharp_helpers.cc (i.e. has the function changed there)? We should probably be clear about which methods have to maintain compatibility with protoc and which are for other purposes.
(I would suggest we should extract them to a separate class, too... and add plenty of unit tests.)